### PR TITLE
feat: #420 implement blacklist to disable unsafe SSL/TLS protocol ver…

### PR DESCRIPTION
**Pull request by silkimen**
…sions on Android

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 3.2.0
+
+- Feature #420: implement blacklist feature to disable SSL/TLS versions on Android (thanks mobisys Mobile Informationssysteme GmbH)
```
Comment (robin-hartmann) [inline CHANGELOG.md:5]: What about:

```suggestion
- Feature #420: implement blacklist feature to disable SSL/TLS versions on Android (thanks to @MobisysGmbH)
```

File: plugin.xml [hunk 2]
```diff
@@ -8,6 +8,7 @@
     <engine name="cordova" version=">=4.0.0"/>
   </engines>
   <dependency id="cordova-plugin-file" version=">=2.0.0"/>
+  <preference name="AndroidBlacklistTlsProtocols" default="SSLv3,TLSv1"/>
```
Comment (robin-hartmann) [inline plugin.xml:11]: Could we call this `AndroidBlacklistSecureSocketProtocols`, so it's more general (like in the [Android docs](https://developer.android.com/reference/javax/net/ssl/SSLSocket#default-configuration-for-different-android-versions))? It doesn't only apply to TLS and maybe in the future there will be another protocol besides TLS and SSL.

File: src/android/com/silkimen/cordovahttp/CordovaHttpPlugin.java [hunk 3]
```diff
@@ -47,6 +47,13 @@ public void initialize(CordovaInterface cordova, CordovaWebView webView) {
 
       this.tlsConfiguration.setHostnameVerifier(null);
       this.tlsConfiguration.setTrustManagers(tmf.getTrustManagers());
+
+      if (this.preferences.contains("androidblacklisttlsprotocols")) {
```
Comment (robin-hartmann) [inline src/android/com/silkimen/cordovahttp/CordovaHttpPlugin.java:51]: I guess accessing preferences isn't case sensitive?

File: test/e2e-specs.js [hunk 4]
```diff
@@ -104,9 +104,17 @@ const helpers = {
 
     return buffer;
   },
+  isTlsBlacklistSupported: function () {
+    if (window.cordova && window.cordova.platformId === 'android') {
+      return true;
+    }
+
+    return false;
```
Comment (robin-hartmann) [inline test/e2e-specs.js:112]: Why not just:

```suggestion
    return window.cordova && window.cordova.platformId === 'android';
```

**Review by robin-hartmann — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (robin-hartmann) [reply CHANGELOG.md:5]: @MobisysGmbH

**Comment by robin-hartmann** [comment]
Also, the Android e2e tests are still failing, even though I just re-ran them.

File: src/android/com/silkimen/cordovahttp/CordovaHttpPlugin.java (hunk 3 above)
Comment (silkimen) [reply src/android/com/silkimen/cordovahttp/CordovaHttpPlugin.java]: Actually it is case-sensitive, but the key is normalized to lower-case string.

**Comment by silkimen** [comment]
> Also, the Android e2e tests are still failing, even though I just re-ran them.

Yes, I'll need to fix this. But strangely enough this seems to happen only on Browserstack, not on SauceLabs and on my local machine. I think it's oaky to merge this for now and then later check for this issue.

**Pull request merged**